Create user model class

vicayala82 opened the pull request:
## Summary
-Created User model class
    -Created methods for add and remove book of collection
    -Created model relationship user whit book
    -Created BookAlreadyOwnedException class
## Trello Card
https://trello.com/c/2fqH75mu/19-crear-modelo-de-user

Commit:
-Created User model class
-Created methods for add and remove book of collection
-Created model relationship user whit book
-Created BookAlreadyOwnedException class
card https://trello.com/c/PycY2Bx1/47-javadoc

juandbc commented on training/src/main/java/wolox/training/models/User.java:
```diff
@@ -0,0 +1,87 @@
+@Entity (name = "users")
+public class User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
```
```suggestion
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
```

juandbc commented on training/src/main/java/wolox/training/exceptions/BookAlreadyOwnedException.java:
```diff
@@ -0,0 +1,25 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(code = HttpStatus.BAD_REQUEST, reason = "Actor Not Found")
```
This should return a conflict a status

juandbc commented on training/src/main/java/wolox/training/exceptions/BookAlreadyOwnedException.java:
```diff
@@ -0,0 +1,25 @@
+    }
+
+    public BookAlreadyOwnedException(String message) {
+        super(message);
```
Like the other exceptions, you can leave the message here

vicayala82 replied:
I fixed it new commit

juandbc commented on training/src/main/java/wolox/training/models/User.java:
```diff
@@ -0,0 +1,87 @@
+
+    public void setId(long id) {
+        this.id = id;
+    }
```
```suggestion
```

juandbc commented on training/src/main/java/wolox/training/models/User.java:
```diff
@@ -0,0 +1,87 @@
+    @Column(nullable = false)
+    private LocalDate birthdate;
+
+    @OneToMany(mappedBy = "user", cascade = CascadeType.ALL)
```
In the most scenarios, it is better to use the merge cascade. Thus, you avoid to delete data and relationships
```suggestion
    @OneToMany(mappedBy = "user", cascade = CascadeType.MERGE)
```

vicayala82 committed:
Update training/src/main/java/wolox/training/models/User.java

Co-authored-by: Juan David Bermudez Celedon <juanbermucele@gmail.com>

vicayala82 committed:
Update training/src/main/java/wolox/training/models/User.java

Co-authored-by: Juan David Bermudez Celedon <juanbermucele@gmail.com>

vicayala82 committed:
Update training/src/main/java/wolox/training/models/User.java

Ok, thanks

Co-authored-by: Juan David Bermudez Celedon <juanbermucele@gmail.com>

Commit:
Resolved conflict in Merge an fixed recomendations

Commit:
Merge branch 'create-user-model-class' of https://github.com/vicayala82/vda-java into create-user-model-class

Commit:
-Fixed with recomendations
-merge and resolve conflicts
card https://trello.com/c/2fqH75mu/19-crear-modelo-de-user

juandbc committed:
Merge branch 'main' into create-user-model-class
